sql/parser/ast.go

```diff
@@ -205,22 +308,33 @@ const (
 	astDropTable       = "DROP TABLE"
 	astDropView        = "DROP VIEW"
 	astRenameTable     = "RENAME TABLE"
+	astShowDatabases   = "SHOW DATABASES"
 	astShowTables      = "SHOW TABLES"
 	astShowIndex       = "SHOW INDEX FROM"
 	astShowColumns     = "SHOW COLUMNS FROM"
 	astShowFullColumns = "SHOW FULL COLUMNS FROM"
 	astTruncateTable   = "TRUNCATE TABLE"
+	astIfNotExists     = " IF NOT EXISTS"
+	astUnique          = "UNIQUE"
+	astPrimaryKey      = "PRIMARY KEY"
+	astKey             = "KEY"
+	astUnsigned        = "UNSIGNED"
 )
 
 func (node *DDL) String() string {
 	switch node.Action {
-	case astCreateIndex, astDropIndex:
+	case astDropIndex:
 		return fmt.Sprintf("%s %s ON %s", node.Action, node.Name, node.NewName)
 	case astRenameTable:
 		return fmt.Sprintf("%s %s %s", node.Action, node.Name, node.NewName)
-	case astCreateDatabase, astCreateTable, astCreateView:
-		return fmt.Sprintf("%s %s", node.Action, node.NewName)
+	case astCreateView:
+		return fmt.Sprintf("%s%s %s", node.Action, node.IfNotExists, node.NewName)
+	case astShowDatabases:
```

Comment:
@PeterMattis Both Oracle and SQL server use term *schema *to name a
collection of schema objects.  An full qualifier of an Oracle schema object
has form of *schema.object_name@db_link *while SQL server one is
*server_name.database_name.schema_name.object_name.*

A Oracle schema has multiple namespaces within it while one SQL server
schema has only one. The following code runs well on Oracle. Please note a
table , an index and a constraint are all named 'person' and they are
created in one schema.


*create table person ( *
*psn_ID varchar(200) constraint pk_person primary key ,*
*first_name varchar(200),*
*last_name varchar(200) not null,*
*gender varchar(8)*
*)*
*;*

*create index person on person(last_name);*
*alter table person add constraint person check (gender in ('M','F',null));*

*alter table person drop constraint person;*

*drop index person;*
*drop table person;*



The reason I mention the above is that data migrations
between heterogeneous systems are always challenging.  A design decision
 which make data migration easier would greatly encourages customers to
adopt cockroach in the future.

Thank you,

Jusong Chen




On Mon, Jun 22, 2015 at 11:58 AM, Peter Mattis <notifications@github.com>
wrote:

> In sql/parser/ast.go
> <https://github.com/cockroachdb/cockroach/pull/1430#discussion_r32970342>:
>
> >  		return fmt.Sprintf("%s %s ON %s", node.Action, node.Name, node.NewName)
> >  	case astRenameTable:
> >  		return fmt.Sprintf("%s %s %s", node.Action, node.Name, node.NewName)
> > -	case astCreateDatabase, astCreateTable, astCreateView:
> > -		return fmt.Sprintf("%s %s", node.Action, node.NewName)
> > +	case astCreateView:
> > +		return fmt.Sprintf("%s%s %s", node.Action, node.IfNotExists, node.NewName)
> > +	case astShowDatabases:
>
> @jusongchen <https://github.com/jusongchen> Cockroach will have the
> concept of a group of tables. Whether that is called a database, schema,
> catalog, namespace, directory or something else is still up in the air. The
> term database certainly has different meanings between DBs. Catalog feels
> like an obscure term to me. Directory is compelling due to the similarity
> to a filesystem directory.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1430/files#r32970342>.
>



-- 
Jusong Chen
408 315 2288